src/lib/stitching/causality/stitching.ts

```diff
@@ -118,16 +118,50 @@ export const causalityStitchingEnvironment = ({
           // (eg first, after, last, before) are forwarded automatically, so we only
           // need the userId.
           resolve: (parent, _args, context, info) => {
-            return info.mergeInfo.delegateToSchema({
-              schema: causalitySchema,
-              operation: "query",
-              fieldName: "_unused_auctionsLotStandingConnection",
-              args: {
-                userId: parent.internalID,
-              },
-              context,
-              info,
-            })
+            return info.mergeInfo
+              .delegateToSchema({
+                schema: causalitySchema,
+                operation: "query",
+                fieldName: "_unused_auctionsLotStandingConnection",
+                args: {
+                  userId: parent.internalID,
+                },
+                context,
+                info,
+              })
+              .then(async (lotStandingsConnection) => {
+                throw new Error("BAD!")
+                console.log("standing connection", lotStandingsConnection)
+                const promisedSaleArtworks = lotStandingsConnection.edges.map(
+                  ({ node: { lot } }) => {
+                    return context
+                      .saleArtworkRootLoader(lot.internalID)
+                      .catch(() => null)
+                  }
+                )
+
+                const availableSaleArtworks = (
+                  await Promise.all(promisedSaleArtworks)
+                ).filter((sa) => sa !== null)
+
+                const availableEdges = lotStandingsConnection.edges.reduce(
```

Comment:
I know this is WIP so I won't comment on the extra logs, etc.!

But overall the approach of querying for lot standings from causality, removing entries where the sale artwork doesn't exist (which we have to determine from gravity) and then returning the results seems 👍.

It is annoying that we have to query one-by-one for the sale artworks 😅. A stop-gap solution could be to add an `ids` param to gravity (I don't think one exists yet) so we can easily batch that request. Definitely not blocking for this, though!

Reply:
Yea, definite 👍  on adding a better first-class support if there's a need to do a batch check on if a group of artworks/sale artworks are published.

Reply:
When we were initially pairing on this we called this out as a definite improvement. This could probably use the batch loader treatment like sales already have.